distributed/protocol/core.py

```diff
@@ -104,7 +105,36 @@ def _encode_default(obj):
             else:
                 return msgpack_encode_default(obj)
 
-        frames[0] = msgpack.dumps(msg, default=_encode_default, use_bin_type=True)
+        try:
+            frames[0] = msgpack.dumps(msg, default=_encode_default, use_bin_type=True)
+        except TypeError as e:
+            logger.warning(e)
+
+            def _encode_default_safe(obj):
+                if isinstance(obj, (Serialize, Serialized)):
+                    offset = len(frames)
+                    frames.extend(create_serialized_sub_frames(obj))
+                    return {"__Serialized__": offset}
+                elif isinstance(obj, (ToPickle, Pickled)):
+                    offset = len(frames)
+                    frames.extend(create_pickled_sub_frames(obj))
+                    return {"__Pickled__": offset}
+                else:
+                    encoded = msgpack_encode_default(obj)
+                    if encoded is not obj or _is_msgpack_serializable(obj):
+                        return encoded
+                    obj = ToPickle(obj)
+                    offset = len(frames)
+                    frames.extend(create_pickled_sub_frames(obj))
+                    return {"__Pickled__": offset}
```

Comment:
in the interest of DRY

```suggestion
                rv = _enode_default(obj)
                if rv is obj and not _is_msgpack_serializable(obj):
                    obj = ToPickle(obj)
                    offset = len(frames)
                    frames.extend(create_pickled_sub_frames(obj))
                    return {"__Pickled__": offset}
                return rv
```

Reply:
Good point